unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/AvatarNamesHUD/AvatarNamesTracker.cs

```diff
@@ -0,0 +1,99 @@
+            name = nameRect.GetComponent<TextMeshProUGUI>();
+            voiceChatCanvasGroup = voiceChatRect.GetComponent<CanvasGroup>();
+            voiceChatAnimator = voiceChatRect.GetComponent<Animator>();
+        }
+
+        public void SetVisibility(bool visible)
+        {
+            visibility = visible;
+            background.gameObject.SetActive(visibility);
+            name.gameObject.SetActive(visibility);
+            voiceChatCanvasGroup.gameObject.SetActive(visibility && (playerStatus?.isTalking ?? false));
+        }
+
+        public void SetPlayerStatus(PlayerStatus newPlayerStatus)
+        {
+            playerStatus = newPlayerStatus;
+            if (playerStatus == null)
+                return;
+
+            name.text = newPlayerStatus.name;
+            name.ForceMeshUpdate(); //To get the new bounds
+            UpdatePosition();
+        }
+
+        public void UpdatePosition()
+        {
+            if (mainCamera == null)
+                mainCamera = mainCamera = Camera.main;
+
+            if (playerStatus == null || mainCamera == null)
+                return;
+
+            Vector3 screenPoint = mainCamera == null ? Vector3.zero : mainCamera.WorldToViewportPoint(playerStatus.worldPosition + OFFSET);
+            float alpha = screenPoint.z < 0 ? 0 : 1.0f + (1.0f - (screenPoint.z / NAME_VANISHING_POINT_DISTANCE));
+
+            if (screenPoint.z > 0)
+            {
+                screenPoint.Scale(canvasRect.rect.size);
+                name.rectTransform.anchoredPosition = screenPoint;
+                background.rectTransform.anchoredPosition = screenPoint;
+                background.rectTransform.sizeDelta = new Vector2(name.textBounds.extents.x * 2.5f, 30);
+                Vector2 voiceChatOffset = -Vector2.Scale(Vector2.right, background.rectTransform.sizeDelta) * 0.5f;
+                (voiceChatCanvasGroup.transform as RectTransform).anchoredPosition = background.rectTransform.anchoredPosition + voiceChatOffset;
+
+                voiceChatCanvasGroup.alpha = alpha;
+                name.color = new Color(name.color.r, name.color.g, name.color.b, alpha);
+                backgroundCanvasGroup.alpha = alpha;
+                voiceChatCanvasGroup?.gameObject.SetActive(visibility && (playerStatus?.isTalking ?? false));
+                voiceChatAnimator.SetBool(VOICE_CHAT_ANIMATOR_TALKING, playerStatus.isTalking);
+
+                background.gameObject.SetActive(visibility);
+                name.gameObject.SetActive(visibility);
+                voiceChatCanvasGroup.gameObject.SetActive(visibility && (playerStatus?.isTalking ?? false));
+            }
+            else
+            {
+                background.gameObject.SetActive(false);
+                name.gameObject.SetActive(false);
+                voiceChatCanvasGroup.gameObject.SetActive(false);
+            }
+        }
+
+        public void DestroyUIElements()
+        {
+            Object.Destroy(background.gameObject);
```

Comment:
Shouldn't we use `GameObject.Destroy `instead of `Object `in unity objects?

Reply:
`Destroy` is a member of `Object`. `GameObject` has it because inherits `Objects`.

Reply:
Yes, but Unity uses an implementation where he passes the time to be destroyed, not sure if it makes any change though

Reply:
I mean, `GameObject` doesn't override `Destroy`. It's the same call